crypto/init.c

```diff
@@ -433,7 +421,6 @@ void OPENSSL_cleanup(void)
     stop_handlers = NULL;
 
     CRYPTO_THREAD_lock_free(init_lock);
-    init_lock = NULL;
```

Comment:
Is-it necessary to remove this assignment?

Reply:
Probably not, but I don't think OPENSSL_init_crypto() is portably re-callable after OPENSSL_cleanup(), so it's unclear that it's worth caring one way or the other.

Reply:
It wouldn't harm to keep this line. However, since the line was introduced in #4294, it's legitimate to remove it if one wants to get as close as possible to a perfect inverse change set.

Reply:
That line should be kept - after a lock_free we should place init_lock back into a cleared (NULL) state.